### PR TITLE
chore: [sinatra] include a web server in example so it boots without changes

### PR DESCRIPTION
**Pull request by pollosp**
Hi, 
I always like examples working by default , so sending you easy "patch" for booting the example without modifying the example into the repo. 
 
 Solves:
 
 ```
 ./start_server.sh 
I, [2023-04-11T17:13:34.785064 #15]  INFO -- : Instrumentation: OpenTelemetry::Instrumentation::Sinatra was successfully installed with the following options {}
/usr/local/bundle/gems/rack-2.2.6.4/lib/rack/handler.rb:45:in `pick': Couldn't find handler for: thin, falcon, puma, HTTP, webrick. (LoadError)
        from /usr/local/bundle/gems/sinatra-3.0.6/lib/sinatra/base.rb:1526:in `run!'
        from ./server.rb:45:in `<class:App>'
        from ./server.rb:20:in `<main>'
root@2419e60e122c:/data# ls
 ```

### CODE REVIEW

**Comment by robbkidd** [comment]
@pollosp Yep! This does solve that problem. Would you be willing to run through the CLA signature process so the project can accept your changes?

**Comment by pollosp** [comment]
Thanks @robbkidd , looks I have issues with EasyCLA as I already signed the document. 
I have opened a ticket and I'm waiting for the response as will be nice to do more contributions in the future.

File: instrumentation/sinatra/example/Gemfile [hunk 1]
```diff
@@ -5,7 +5,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 source 'https://rubygems.org'
-
+gem 'puma'
```
Comment (arielvalentin) [inline instrumentation/sinatra/example/Gemfile:8]: Does webrick not work by default?

Comment (arielvalentin) [reply instrumentation/sinatra/example/Gemfile:8]: Yeah I guess I am surprised that it does not work and it should probably use whatever is bundled in the Ruby stdlib, which is what I thought sinatra used by default? but I guess not. 

Puma should be fine.

**Review by arielvalentin — APPROVED** [review]

**Review by robbkidd — APPROVED** [review]
Nancy Sinatra approves of the boot.

**Comment by robbkidd** [comment]
@pollosp thank you for making life a little easier for folks trying things out! 💞

**Pull request merged**